### ISSUE TITLE
bug: test-connection provides false positive for plugins where it is not implemented

### ISSUE THREAD

**Issue by jschafer-kp** [cloudquery/cloudquery#19338]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When I call `test-connection` on the [Microsoft Entra ID plugin](https://hub.cloudquery.io/plugins/source/cloudquery/entraid/latest/docs), it returns a successful message
```
Test Connection Results
source       entraid (cloudquery/entraid@v1.4.3)  Success
```
However, when I look at the logs, I see this error message
```
2024-10-08T15:05:22Z ERR finished call grpc.code=Unimplemented grpc.component=server grpc.error="rpc error: code = Unimplemented desc = TestConnection feature is not implemented in this plugin" grpc.method=TestConnection grpc.method_type=unary grpc.service=cloudquery.plugin.v3.Plugin grpc.start_time=2024-10-08T10:05:22-05:00 grpc.time_ms=1.367 invocation_id=<removed> module=cli peer.address= protocol=grpc
```

### Expected Behavior

I would expect that the call would show an error or message indicating that the feature has not been implemented, even for preview plugins.

### CloudQuery (redacted) config

```
kind: source
# Common source-plugin configuration
spec:
  name: entraid
  path: cloudquery/entraid
  registry: cloudquery
  version: "v1.3.2"
  tables: ["entraid_users"]
  destinations: ["postgresql"]

  # Entra ID specific configuration
  spec:
    # Optional parameters
    # concurrency: 50000

```

### Steps To Reproduce

1. With the above config
2. Run `test-connection <config file name>
3. See successful message on output
4. See error in logs

### CloudQuery (redacted) logs

```
2024-10-08T16:09:59Z INF Loading spec(s) args=["entraid.yml"] invocation_id=6c646d1d-b6bc-4506-80bb-592bf5d44548 module=cli
2024-10-08T16:10:00Z INF Plugin server listening address=/var/folders/l6/0d71k0vj4yqg69ck51fn_5440000gn/T/cq-dTARmNQnCKYnbcje.sock invocation_id=6c646d1d-b6bc-4506-80bb-592bf5d44548 module=cli
2024-10-08T16:10:01Z INF Testing source invocation_id=6c646d1d-b6bc-4506-80bb-592bf5d44548 module=cli source="entraid (cloudquery/entraid@v1.4.3)"
2024-10-08T16:10:01Z INF started call grpc.component=server grpc.method=TestConnection grpc.method_type=unary grpc.service=cloudquery.plugin.v3.Plugin grpc.start_time=2024-10-08T11:10:01-05:00 grpc.time_ms=1.825 invocation_id=6c646d1d-b6bc-4506-80bb-592bf5d44548 module=cli peer.address= protocol=grpc
2024-10-08T16:10:01Z ERR finished call grpc.code=Unimplemented grpc.component=server grpc.error="rpc error: code = Unimplemented desc = TestConnection feature is not implemented in this plugin" grpc.method=TestConnection grpc.method_type=unary grpc.service=cloudquery.plugin.v3.Plugin grpc.start_time=2024-10-08T11:10:01-05:00 grpc.time_ms=2.051 invocation_id=6c646d1d-b6bc-4506-80bb-592bf5d44548 module=cli peer.address= protocol=grpc
2024-10-08T16:10:01Z INF started call grpc.component=server grpc.method=Init grpc.method_type=unary grpc.service=cloudquery.plugin.v3.Plugin grpc.start_time=2024-10-08T11:10:01-05:00 grpc.time_ms=0.005 invocation_id=6c646d1d-b6bc-4506-80bb-592bf5d44548 module=cli peer.address= protocol=grpc
2024-10-08T16:10:01Z INF finished call grpc.code=OK grpc.component=server grpc.method=Init grpc.method_type=unary grpc.service=cloudquery.plugin.v3.Plugin grpc.start_time=2024-10-08T11:10:01-05:00 grpc.time_ms=183.939 invocation_id=6c646d1d-b6bc-4506-80bb-592bf5d44548 module=cli peer.address= protocol=grpc
2024-10-08T16:10:01Z INF Test connection completed invocation_id=6c646d1d-b6bc-4506-80bb-592bf5d44548 module=cli testresults=[{"plugin_kind":"source","plugin_ref":"entraid (cloudquery/entraid@v1.4.3)","success":true}]
2024-10-08T16:10:01Z INF waiting for source plugin to terminate invocation_id=6c646d1d-b6bc-4506-80bb-592bf5d44548 module=cli
2024-10-08T16:10:01Z INF Got stop signal. Plugin server shutting down address=/var/folders/l6/0d71k0vj4yqg69ck51fn_5440000gn/T/cq-dTARmNQnCKYnbcje.sock invocation_id=6c646d1d-b6bc-4506-80bb-592bf5d44548 module=cli signal=interrupt
```

### CloudQuery version

cloudquery version 6.8.2

### Pull request (optional)

- [ ] I can submit a pull request

**Comment by marianogappa** [comment]
Hi @jschafer-kp! Sorry, the automated workflow closed the issue, but the fix hasn't been released yet. Thank you for reporting! Hopefully this will be fixed in v6.8.4: https://github.com/cloudquery/cloudquery/pull/19354

**Comment by erezrokah** [comment]
Released in https://github.com/cloudquery/cloudquery/releases/tag/cli-v6.8.4 thanks @jschafer-kp for reporting and @marianogappa for fixing
